sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -301,30 +301,37 @@ export class ManagementClient extends LinkEntity<RequestResponseLink> {
     internalLogger: ServiceBusLogger,
     sendRequestOptions: SendManagementRequestOptions = {}
   ): Promise<RheaMessage> {
+    if (request.message_id == undefined) {
+      request.message_id = generate_uuid();
+    }
     const retryTimeoutInMs =
       sendRequestOptions.timeoutInMs ?? Constants.defaultOperationTimeoutInMs;
     const initOperationStartTime = Date.now();
-    const actionAfterTimeout = () => {
+    const actionAfterTimeout = (reject: (reason?: any) => void) => {
       const desc: string = `The request with message_id "${request.message_id}" timed out. Please try again later.`;
       const e: Error = {
         name: "OperationTimeoutError",
         message: desc
       };
 
-      throw e;
+      reject(e);
     };
 
-    const waitTimer = setTimeout(actionAfterTimeout, retryTimeoutInMs);
-
-    internalLogger.verbose(`${this.logPrefix} Acquiring lock to get the management req res link.`);
+    await new Promise<void>(async (resolve, reject) => {
+      const waitTimer = setTimeout(() => actionAfterTimeout(reject), retryTimeoutInMs);
+      internalLogger.verbose(
+        `${this.logPrefix} Acquiring lock to get the management req res link.`
+      );
```

Comment:
Ah, I was thinking something more like this:
```ts
let waitTimer;
const operationTimeout= new Promise((_, reject) => {
  waitTimer = setTimeout(() => actionAfterTimeout(reject), retryTimeoutInMs);
});

try {
  if (!this.isOpen()) {
    await Promise.race([
      this._init(sendRequestOptions?.abortSignal,
      operationTimeout
    ]);
  }
} finally {
  clearTimeout(waitTimer);
}
```

eslint will complain if you pass an async function to a promise constructor. I actually think something like this is a bit more clear on the intent since you can see how the timeout relates to the init call.

Reply:
"race" looks nice, I'll update, thanks!

Reply:
> eslint will complain if you pass an async function to a promise constructor

I see.. https://eslint.org/docs/rules/no-async-promise-executor

Reply:
Updated.